zeroconf/__init__.py

```diff
@@ -2444,13 +2449,22 @@ def handle_response(self, msg: DNSIncoming) -> None:
             updated = True
 
             if record.unique:  # https://tools.ietf.org/html/rfc6762#section-10.2
-                for entry in self.cache.entries():
-
-                    if entry == record:
-                        updated = False
-
-                    if DNSEntry.__eq__(entry, record) and (record.created - entry.created > 1000):
-                        self.cache.remove(entry)
+                # Since the cache format is keyed on DNSEntry
+                # we can avoid iterating everything in the cache and
+                # only look though entries for the specific DNSEntry
+                # cache key.
+                entries = self.cache.entries_with_name(DNSEntry(record.name, record.type, record.class_).key)
```

Comment:
`DNSEntry(record.name, record.type, record.class_).key` just returns `record.name.lower()`, but `entries_with_name()` already performs `lower()` on the name passed, so `DNSEntry(record.name, record.type, record.class_).key` should be just `record.name` I think.

Reply:
```
2020-03-07 11:44:27 WARNING (zeroconf-Engine) [zeroconf] RECORD CHECK: dnsentry_key=[wiced-hap-3e2734.local.] name=[WICED-hap-3E2734.local.]
2020-03-07 11:44:27 WARNING (zeroconf-Engine) [zeroconf] RECORD CHECK: dnsentry_key=[wiced-hap-3e2734.local.] name=[WICED-hap-3E2734.local.]
```

Confirmed